### PR TITLE
Avoid NRE in tenant creation

### PR DESCRIPTION
**Pull request by hishamco**
While I'm working on other PR, I just found a NRE when I create a new tenant, this is because the feature profles is `null`

### CODE REVIEW

**Comment by jtkech** [comment]
@hishamco @deanmarcussen

Yes I can confirm, if initially there is no `FeatureProfiles` in the `Create / Edit` post actions the `FeatureProfiles` of the bounded `EditTenantViewModel model` is null, so if it fails to validate we do `return View(model)` to re-display the form and then we get an NRE in `Create / Edit.cshtml` on `Model.FeatureProfiles.Any()`.

So maybe we not only need a null check in `Create.cshtml` but also in `Edit.cshtml`.

**Comment by hishamco** [comment]
Thanks @jtkech , frankly I didn't check the edit view but seems you confirm that NRE occurs

**Comment by jtkech** [comment]
@hishamco 

Look at my updated comment above, I think it's beter to re-init the `FeatureProfiles` list in place of adding null checks.

**Comment by hishamco** [comment]
> Look at my updated comment above, I think it's beter to re-init the FeatureProfiles list in place of adding null checks.

I already suggested that too ;) I will update the PR

**Comment by jtkech** [comment]
@hishamco 

See the updated part of my previous comment

> Or better I think, in the post actions, if it fails to validate we need to re-populate the FeatureProfiles list, as we do in the related get actions, even if it ends up in an empty list (but not null).

In fact I was thinking about re-populating the FeatureProfiles list, as it is done in the related get actions, not just init it as an empty list through a new ctor.

File: src/OrchardCore.Modules/OrchardCore.Tenants/Controllers/AdminController.cs [hunk 1]
```diff
@@ -329,11 +324,10 @@ public async Task<IActionResult> Create(EditTenantViewModel model)
                 await ValidateViewModel(model, true);
             }
 
+            // Creates a default shell settings based on the configuration.
+            var shellSettings = _shellSettingsManager.CreateDefaultSettings();
```
Comment (deanmarcussen) [inline src/OrchardCore.Modules/OrchardCore.Tenants/Controllers/AdminController.cs:328]: don't move this

File: src/OrchardCore.Modules/OrchardCore.Tenants/Controllers/AdminController.cs [hunk 2]
```diff
@@ -356,6 +350,7 @@ public async Task<IActionResult> Create(EditTenantViewModel model)
             var recipeCollections = await Task.WhenAll(_recipeHarvesters.Select(x => x.HarvestRecipesAsync()));
             var recipes = recipeCollections.SelectMany(x => x).Where(x => x.IsSetupRecipe).OrderBy(r => r.DisplayName).ToArray();
             model.Recipes = recipes;
+            model.FeatureProfiles = await GetFeatureProfilesAsync(shellSettings["FeatureProfile"]);
```
Comment (deanmarcussen) [inline src/OrchardCore.Modules/OrchardCore.Tenants/Controllers/AdminController.cs:353]:
```suggestion
            model.FeatureProfiles = await GetFeatureProfilesAsync(model.FeatureProfile);
```

File: src/OrchardCore.Modules/OrchardCore.Tenants/Controllers/AdminController.cs [hunk 3]
```diff
@@ -485,6 +476,7 @@ public async Task<IActionResult> Edit(EditTenantViewModel model)
...
```
Comment (deanmarcussen) [inline src/OrchardCore.Modules/OrchardCore.Tenants/Controllers/AdminController.cs:479]:
```suggestion
            model.FeatureProfiles = await GetFeatureProfilesAsync(model.FeatureProfile);
```

**Review by deanmarcussen — COMMENTED** [review]
ok, so from what @jtkech says and I can see you identified an issue when the form is invalid.
So the suggestion is to use the values from the model, as it was presented on the form post.

**Review by deanmarcussen — APPROVED** [review]

**Pull request merged**